static/app/views/issueList/groupSearchViewTabs/draggableTabBar.tsx

```diff
@@ -188,17 +192,24 @@ export function DraggableTabBar({
         },
         ...tabs.slice(idx + 1),
       ];
+      navigate({
+        query: {
+          ...queryParams,
+          viewId: tempId,
+        },
+        pathname: `/organizations/${orgSlug}/issues/`,
+      });
       setTabs(newTabs);
-      setSelectedTabKey(tempId);
+      tabListState?.setSelectedKey(tempId);
```

Comment:
I would have expected `tabListState?.setSelectedKey(tempId);` to do the navigation automatically, but it does not for some reason. I suspect this might be something to do with the fact that the new tab has not been registered into tablist state yet. For now, this is a good workaround that achieves the expected behavior.